### PR TITLE
refactor: refactor security config field

### PR DESCRIPTION
**Pull request by khalid-jobs**

### CODE REVIEW

File: docs/advanced/security.md [hunk 1]
```diff
@@ -44,10 +44,25 @@ data:
...
+Assume that the ip of cloudcore service is 110.8.52.21, the installation commands are as follows:
+```yaml
+helm install edgemesh --set server.nodeName=dev-02 \
+--set "server.advertiseAddress={109.8.58.38}" \
+--set server.modules.tunnel.security.enable=true \
+--set server.modules.tunnel.security.httpServer="https://110.8.52.21:10002" \
+--set server.image=kubeedge/edgemesh-server:latest \
+--set agent.modules.tunnel.security.enable=true \
+--set agent.modules.tunnel.security.httpServer="https://110.8.52.21:10002" \
+--set agent.image=kubeedge/edgemesh-agent:latest  .
```
Comment (Poorunga) [inline docs/advanced/security.md:67]: Why not use online package?

**Comment by Poorunga** [comment]
need to update gateway

File: build/helm/edgemesh/charts/agent/templates/04-configmap.yaml [hunk 2]
```diff
@@ -65,3 +65,6 @@ data:
       tunnel:
         enable: {{ .Values.modules.tunnel.enable }}
         listenPort: {{ .Values.modules.tunnel.listenPort }}
+        security:
+          enable: {{ .Values.modules.tunnel.security.enable }}
+          httpServer: {{ .Values.modules.tunnel.security.httpServer }}
```
Comment (Poorunga) [inline build/helm/edgemesh/charts/agent/templates/04-configmap.yaml:70]: why not also add tlsCAFile, tlsPrivateKeyFile, tlsCertFile?

File: agent/pkg/tunnel/config/config.go [hunk 3]
```diff
@@ -14,27 +14,24 @@ type TunnelAgentConfig struct {
 	// if set to false (for debugging etc.), skip checking other TunnelAgent configs.
 	// default false
 	Enable bool `json:"enable,omitempty"`
-	// TunnelACLConfig indicates the set of tunnel agent config about acl
-	TunnelACLConfig acl.TunnelACLConfig `json:"ACL,omitempty"`
+	// Security indicates the set of tunnel agent config about security
+	Security security.Security `json:"security,omitempty"`
```
Comment (Poorunga) [inline agent/pkg/tunnel/config/config.go:18]: Is the pointer field better?

File: build/helm/edgemesh/charts/agent/values.yaml [hunk 4]
```diff
@@ -34,3 +34,7 @@ modules:
   tunnel:
     enable: true
     listenPort: 20006
+    security:
+      enable: false
+      httpServer: ""
```
Comment (Poorunga) [inline build/helm/edgemesh/charts/agent/values.yaml:39]: why not also add tlsCAFile, tlsPrivateKeyFile, tlsCertFile?

File: docs/zh/reference/config-items.md [hunk 5]
```diff
@@ -268,19 +268,19 @@ modules:
...
+#### 表1-4-4-1: security
 
 | 名称 | 类型 | 默认值 | 描述 |
 | ---- | ---- | ---- | ---- |
+| enable | bool | false | 子模块启动开关 |
 | tlsCaFile | string | /etc/kubeedge/edgemesh/agent/acls/rootCA.crt | CA 文件路径 |
 | tlsCertFile | string | /etc/kubeedge/edgemesh/agent/acls/server.crt | 证书文件路径 |
 | tlsPrivateKeyFile | string | /etc/kubeedge/edgemesh/agent/acls/server.key | 私钥文件路径 |
-| token | string | 无 | 口令。等同于 edgecore.yaml 中的 token |
+| token | string | 这个值会通过挂载kubeedge namespace下面的`tokensecret` secret 自动获取 | 口令。等同于 edgecore.yaml 中的 token |
```
Comment (Poorunga) [inline docs/zh/reference/config-items.md:283]:
```suggestion
| token | string | 无 | 等同于 edgecore.yaml 中的 token。禁止用户手动配置，这个值会通过挂载kubeedge namespace下面的`tokensecret` secret 自动获取 |
```

File: docs/advanced/security.md (hunk 1 above)
Comment (khalid-jobs) [reply docs/advanced/security.md]: fix

File: build/helm/edgemesh/charts/agent/templates/04-configmap.yaml (hunk 2 above)
Comment (khalid-jobs) [reply build/helm/edgemesh/charts/agent/templates/04-configmap.yaml:70]: fix

File: agent/pkg/tunnel/config/config.go (hunk 3 above)
Comment (khalid-jobs) [reply agent/pkg/tunnel/config/config.go]: fix

File: build/helm/edgemesh/charts/agent/values.yaml (hunk 4 above)
Comment (khalid-jobs) [reply build/helm/edgemesh/charts/agent/values.yaml:39]: fix

File: docs/zh/reference/config-items.md (hunk 5 above)
Comment (khalid-jobs) [reply docs/zh/reference/config-items.md]: fix

**Pull request merged**